src/main/java/org/reaktivity/nukleus/http/internal/HttpController.java

```diff
@@ -98,7 +98,67 @@ public String name()
         return "http";
     }
 
-    public CompletableFuture<Long> route(
+    public CompletableFuture<Long> routeInputNone(
+            String source,
+            long sourceRef,
+            String target,
+            long targetRef,
+            Map<String, String> headers)
+    {
+        return route(Role.INPUT, State.NONE, source, sourceRef, target, targetRef, headers);
+    }
+
+    public CompletableFuture<Long> routeInputEstablished(
+            String source,
+            long sourceRef,
+            String target,
+            long targetRef,
+            Map<String, String> headers)
+    {
+        return route(Role.INPUT, State.ESTABLISHED, source, sourceRef, target, targetRef, headers);
+    }
+
+    public CompletableFuture<Long> routeInputNew(
+            String source,
+            long sourceRef,
+            String target,
+            long targetRef,
+            Map<String, String> headers)
+    {
+        return route(Role.INPUT, State.NEW, source, sourceRef, target, targetRef, headers);
+    }
+
+    public CompletableFuture<Long> routeOutputNone(
+            String source,
+            long sourceRef,
+            String target,
+            long targetRef,
+            Map<String, String> headers)
+    {
+        return route(Role.OUTPUT, State.NONE, source, sourceRef, target, targetRef, headers);
+    }
+
+    public CompletableFuture<Long> routeOutputEstablished(
+            String source,
+            long sourceRef,
+            String target,
+            long targetRef,
+            Map<String, String> headers)
+    {
+        return route(Role.OUTPUT, State.ESTABLISHED, source, sourceRef, target, targetRef, headers);
+    }
+
+    public CompletableFuture<Long> routeOutputNew(
+            String source,
+            long sourceRef,
+            String target,
+            long targetRef,
+            Map<String, String> headers)
+    {
+        return route(Role.OUTPUT, State.NEW, source, sourceRef, target, targetRef, headers);
+    }
+
+    private CompletableFuture<Long> route(
```

Comment:
Please move now-private route method to end of file, just before now-private unroute method.